libass/ass_fontconfig.c

```diff
@@ -83,6 +83,71 @@ static void destroy(void *priv)
     free(fc);
 }
 
+static int compare_patterns(const FcPattern *p1, const FcPattern *p2)
+{
+    // Temporary workaround for fontconfig family name matching being too lenient.
+    // Ensure regular variants come first, so if the score is tied, they'll be selected.
+    // Weight and slant alread y factor into scoring for regular selection and fallback
+    // selection uses a different list already sorted by FcFontSort; thus just width suffices.
+    FcResult res;
+    int wi1, wi2;
+    res = FcPatternGetInteger(p1, FC_WIDTH, 0, &wi1);
+    res |= FcPatternGetInteger(p2, FC_WIDTH, 0, &wi2);
+    if (res != FcResultMatch)
+        return 0;
+    wi1 = abs(wi1 - FC_WIDTH_NORMAL);
+    wi2 = abs(wi2 - FC_WIDTH_NORMAL);
+    if (wi1 != wi2)
+        return wi1 - wi2;
+
+    return 0;
```

Comment:
Nit: this if/return/return seems silly and could just be `return wi1 - wi2;`

It might be intended to allow adding more sort criteria later, but the comment suggests we shouldn’t ever have any more.

Reply:
yeah, it’s a leftover from other tested versions with more criteria; shortened it now